src/monitoring_service/handlers.py

```diff
@@ -178,24 +191,25 @@ def action_monitoring_triggered_event_handler(event: Event, context: Context):
             ),
             address=context.ms_state.blockchain_state.monitor_contract_address,
         )
-        tx_hash = contract.functions.monitor(
-            monitor_request.signer,
-            monitor_request.non_closing_signer,
-            monitor_request.balance_hash,
-            monitor_request.nonce,
-            monitor_request.additional_hash,
-            monitor_request.signature,
-            monitor_request.non_closing_signature,
-            monitor_request.reward_amount,
-            monitor_request.token_network_address,
-            monitor_request.reward_proof_signature,
-        ).transact(
-            {'gas_limit': 350000},
-            private_key='',  # FIXME: use signing middleware
-        )
-        log.info(f'Submit MR to SC, got tx_hash {tx_hash}')
-        assert tx_hash is not None
-        # TODO: store tx hash in state
+        try:
+            tx_hash = contract.functions.monitor(
+                monitor_request.signer,
+                monitor_request.non_closing_signer,
+                monitor_request.balance_hash,
+                monitor_request.nonce,
+                monitor_request.additional_hash,
+                monitor_request.closing_signature,
+                monitor_request.non_closing_signature,
+                monitor_request.reward_amount,
+                monitor_request.token_network_address,
+                monitor_request.reward_proof_signature,
+            ).transact({'gas': 100_000})  # TODO: estimate gas better here
+            log.info(f'Submit MR to SC, got tx_hash {encode_hex(tx_hash)}')
+            assert tx_hash is not None
+            data = context.w3.eth.waitForTransactionReceipt(tx_hash)
```

Comment:
Is this quick enough to be done with blocking?

Reply:
Ui, I missed that. That should go away.